pkg/apis/apps/v1alpha1/apimanager_types.go

```diff
@@ -473,6 +478,11 @@ func (apimanager *APIManager) setAPIManagerCommonSpecDefaults() bool {
 		changed = true
 	}
 
+	if spec.PodDisruptionBudget == nil {
```

Comment:
For other fields we also followed this approach of initializing the field explicitly when it is not initialized.

This has proved inconvenient for some use cases and currently we tend to favour the approach of not explicitly setting the spec field. The reason for that is that explicitly setting a spec field makes hard to differentiate whether the user has been the one that has set the field or it is a default set by the operator. This is a scenario that we might be interested to check as we have needed for other fields (for example the database field in the System subsection of the CR).

In case we really want to show the state of this we could add it in the "status" section of the CR.

I tend to incline to not explicitly set the default in the spec.

cc @eguzki

Reply:
ok, I am happy to remove this.